lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart

```diff
@@ -13,42 +15,56 @@ import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/
 import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/components/current_user_avatar.dart';
 import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/components/parent_entity.dart';
 import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/components/quoted_entity.dart';
+import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_cover.dart';
+import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/hooks/use_keyboard_scroll_handler.dart';
 import 'package:ion/app/features/feed/views/components/actions_toolbar/actions_toolbar.dart';
 import 'package:ion/app/features/feed/views/components/text_editor/hooks/use_quill_controller.dart';
 import 'package:ion/app/features/feed/views/components/text_editor/text_editor.dart';
 import 'package:ion/app/features/feed/views/components/toolbar_buttons/toolbar_buttons.dart';
+import 'package:ion/app/features/feed/views/components/visibility_settings_toolbar/visibility_settings_toolbar.dart';
 import 'package:ion/app/features/feed/views/pages/cancel_creation_modal/cancel_creation_modal.dart';
 import 'package:ion/app/features/nostr/model/event_reference.dart';
 import 'package:ion/app/router/components/navigation_app_bar/navigation_app_bar.dart';
 import 'package:ion/app/router/components/sheet_content/sheet_content.dart';
 import 'package:ion/app/router/utils/show_simple_bottom_sheet.dart';
 
 class CreatePostModal extends HookConsumerWidget {
-  const CreatePostModal({
+  CreatePostModal({
     required this.showCollapseButton,
     super.key,
     this.parentEvent,
     this.quotedEvent,
     this.content,
+    this.videoPath,
   });
 
   final EventReference? parentEvent;
-
   final EventReference? quotedEvent;
-
   final String? content;
-
   final bool showCollapseButton;
+  final String? videoPath;
+
+  final visibilityToolbarKey = GlobalKey();
```

Comment:
Use hooks to create global keys. Flutter may create another instance of `CreatePostModal` and keys are gonna be recreated.